core/music/SpotifyApi.js

```diff
@@ -0,0 +1,112 @@
+class SpotifyApi {
+    constructor(client) {
+        this.client = client;
+        this.access_token = null;
+        this.client_id = this.client.config.player.key.spotifyClient;
+        this.client_secret = this.client.config.player.key.spotifySecret;
+    }
+
+    async fetchToken() {
+        const data = await fetch(`https://accounts.spotify.com/api/token?grant_type=client_credentials&client_id=${this.client_id}&client_secret=${this.client_secret}`, {
```

Comment:
Fetch will not send an error if response status code is other than 200 => need to check data.statusText